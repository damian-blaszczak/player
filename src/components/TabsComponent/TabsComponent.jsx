import React from 'react';
import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { makeStyles } from '@material-ui/core';
import styled from 'styled-components';

const useStyles = makeStyles({
  customTabIndicator: {
    backgroundColor: '#fff',
    top: 0
  },
});

const TabLabel = styled.span`
  font-size: 1.5rem;
`;

const TabPanelContainer = styled.div`
  max-width: 810px;
  margin: auto;
`;

const TabPanel = (props) => {
  const { children, value, index } = props;
  return value === index && children
}

const TabsComponent = ({
  tabIndex,
  handleChange,
  tabs,
  tabPanels
}) => {
  const classes = useStyles();
  return (
    <>
      <Tabs
        variant="fullWidth"
        value={tabIndex}
        onChange={handleChange}
        classes={{ indicator: classes.customTabIndicator }}
      >
        {tabs.map(tab => (
          <Tab key={tab} label={<TabLabel>{tab}</TabLabel>} />
        ))}
      </Tabs>
      {tabPanels.map((tabPanel, index) => (
        <TabPanelContainer key={index} >
          <TabPanel value={tabIndex} index={index}>
            {tabPanel}
          </TabPanel>
        </TabPanelContainer>
      ))}
    </>
  );
}

TabsComponent.propTypes = {
  tabIndex: PropTypes.number,
  handleChange: PropTypes.func,
  tabs: PropTypes.arrayOf(PropTypes.string),
  tabPanels: PropTypes.arrayOf(PropTypes.node)
};

export default TabsComponent;
