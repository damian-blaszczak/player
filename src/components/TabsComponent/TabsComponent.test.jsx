import React from 'react';
import { render } from '@testing-library/react';
import TabsComponent from './TabsComponent';

it('renders correctly', () => {
  const props = {
    tabIndex: 1,
    handleChange: jest.fn(),
    handleChangeIndex: jest.fn(),
    tabs: ['tab1', 'tab2', 'tab3'],
    tabPanels: [
      <div>1</div>,
      <div>2</div>,
      <div>3</div>
    ]
  };
  const { container } = render(
    <TabsComponent {...props} />
  );
  expect(container).toMatchSnapshot();
});
