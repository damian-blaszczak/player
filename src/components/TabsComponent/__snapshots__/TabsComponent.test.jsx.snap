// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        class="MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <span
              class="sc-AxjAm kPsxnP"
            >
              tab1
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <span
              class="sc-AxjAm kPsxnP"
            >
              tab2
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit MuiTab-fullWidth"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <span
              class="sc-AxjAm kPsxnP"
            >
              tab3
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-2 PrivateTabIndicator-colorSecondary-4 MuiTabs-indicator makeStyles-customTabIndicator-1"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
  <div
    class="sc-AxirZ bZABrH"
  />
  <div
    class="sc-AxirZ bZABrH"
  >
    <div>
      2
    </div>
  </div>
  <div
    class="sc-AxirZ bZABrH"
  />
</div>
`;
